### PR TITLE
[modern-cpp-kafka] New port

### LINKED ISSUE #32897: [New Port Request] <morganstanley/modern-cpp-kafka>
### Library name

morganstanley/modern-cpp-kafka

### Library description

The modern-cpp-kafka API is a layer of C++ wrapper based on librdkafka (the C part only), with high quality, but more friendly to users

### Source repository URL

https://github.com/morganstanley/modern-cpp-kafka

### Project homepage (if different from the source repository)

_No response_

### Anything else that is useful to know when adding (such as optional features the library may have that should be included)

_No response_

### PR DESCRIPTION
**Pull request by xiaozhuai**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

Close #32897

### CODE REVIEW

File: ports/modern-cpp-kafka/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+project(modern-cpp-kafka LANGUAGES CXX)
+
+include(GNUInstallDirs)
+
+find_package(RdKafka CONFIG REQUIRED)
+
+add_library(modern-cpp-kafka INTERFACE)
+target_include_directories(modern-cpp-kafka INTERFACE $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/modern-cpp-kafka>)
+target_link_libraries(modern-cpp-kafka INTERFACE RdKafka::rdkafka)
```
Comment (dg0yt) [inline ports/modern-cpp-kafka/CMakeLists.txt:11]: This needs a `find_dependency`.

**Review by dg0yt — CHANGES_REQUESTED** [review]
I don't think this really needs a CMake project. You can just add a config file directly to the port and install from the portfile.

File: ports/modern-cpp-kafka/CMakeLists.txt (hunk 1 above)
Comment (xiaozhuai) [reply ports/modern-cpp-kafka/CMakeLists.txt]: This has been handled in portfile.cmake

**Comment by xiaozhuai** [comment]
> I don't think this really needs a CMake project. You can just add a config file directly to the port and install from the portfile.

I know. But I prefer provide a CMakeLists.

**Comment by xiaozhuai** [comment]
Usage passed on x64-osx with following demo code

```cmake
set(CMAKE_CXX_STANDARD 17)

find_package(unofficial-modern-cpp-kafka CONFIG REQUIRED)
add_executable(test_kafka main.cpp)
target_link_libraries(test_kafka PRIVATE unofficial::modern-cpp-kafka::modern-cpp-kafka)
```

```cpp
#include "kafka/AdminClient.h"
#include "kafka/Error.h"

int main() {
    kafka::Properties props;
    kafka::clients::admin::AdminClient client(props);
    return 0;
}
```

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-windows` (header files found):
```
modern-cpp-kafka provides CMake targets:

    find_package(unofficial-modern-cpp-kafka CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::modern-cpp-kafka::modern-cpp-kafka)
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**